### PR TITLE
docs: developer theming wording, grammar, typo fixes

### PR DESCRIPTION
**Pull request by adamjohnson**
## What I did

1. Fixed/improved grammar, typos, and wording of Developer Theming docs

## Testing Instructions

1. Read the content on the DP for Developer Theming and make sure it flows nicely.

Related to #1892 and #1897.

### CODE REVIEW

File: docs/theming/developers.md [hunk 1]
```diff
@@ -200,41 +200,41 @@ themable graphics
...
-This is important not only for helping us maintain our own design guidelines, 
-but also for accessibility compliance and enabling great experiences for all our 
+This is important not only for helping us maintain our design guidelines but
+also for accessibility compliance and enabling great experiences for all our 
 users.
 
 The context protocol is enabled by two [reactive controllers][controllers]: the
 [provider controller][providersrc], and the [consumer controller][consumersrc].
-These controllers work in tandem to alternately provide the context data, or
+These controllers work together to provide the context data, or
```
Comment (bennypowers) [inline docs/theming/developers.md:230]: I wasn't sure about "together"  can we really say they "work together to provide and consume"? One provides, the other consumes. They work together to propagate styles to the correct elements, but not to either provide or consume

Comment (adamjohnson) [reply docs/theming/developers.md]: Good call. Check out [1bdecba](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/1bdecba001a4bc4752245dc512bb6858eaac781c) and see if that is an improvement.

File: docs/theming/developers.md [hunk 2]
```diff
@@ -35,24 +35,24 @@ hasToc: true
 
 ## Using the theming system
 
-<abbr title="Red Hat Design System">RHDS</abbr>' theming system is a high-level 
+<abbr title="Red Hat Design System">RHDS</abbr>’ theming system is a high-level
```
Comment (hellogreg) [inline docs/theming/developers.md:38]: `RHDS’` → `RHDS’s `

If I'm reading this right, I think I'd add a colon to this sentence, or at the very least remove the comma after "tokens":

`components of the system: [tokens](https://deploy-preview-1937--red-hat-design-system.netlify.app/tokens/) and [elements](https://deploy-preview-1937--red-hat-design-system.netlify.app/elements/).`

File: docs/theming/developers.md [hunk 3]
```diff
@@ -35,24 +35,24 @@ hasToc: true
 
 ## Using the theming system
 
-<abbr title="Red Hat Design System">RHDS</abbr>' theming system is a high-level 
+<abbr title="Red Hat Design System">RHDS</abbr>’ theming system is a high-level 
 expression of the lower-level components of the system, [tokens][tokens], and 
-[elements][elements], and in turn, it factors into the development of the 
-highest level design system component, [patterns][patterns]. In order to use 
+[elements][elements]. In turn, it factors into the development of the 
+highest-level design system component, [patterns][patterns]. To use
```
Comment (hellogreg) [inline docs/theming/developers.md:41]: If you do add the colon above, add one here, too:

`the highest-level design system component: [patterns][patterns].`

File: docs/theming/developers.md [hunk 4]
```diff
@@ -35,24 +35,24 @@ hasToc: true
...
+and elements. In other words, theming is the developer’s process of 
 orchestrating design tokens with elements, particularly by way of themeable 
-container elements
+container elements.
 
 ### Themeable containers
 
 In <abbr>RHDS</abbr>, special provider elements such as `<rh-surface>`, 
-`<rh-card>`, `<rh-tabs>`, and others are considered themeble containers.  A 
+`<rh-card>`, `<rh-tabs>`, and others are considered themeble containers. A
```
Comment (hellogreg) [inline docs/theming/developers.md:50]: `themeble` → `themeable`

File: docs/theming/developers.md [hunk 5]
```diff
@@ -35,24 +35,24 @@ hasToc: true
...
 orchestrating design tokens with elements, particularly by way of themeable 
-container elements
+container elements.
 
 ### Themeable containers
 
 In <abbr>RHDS</abbr>, special provider elements such as `<rh-surface>`, 
-`<rh-card>`, `<rh-tabs>`, and others are considered themeble containers.  A 
+`<rh-card>`, `<rh-tabs>`, and others are considered themeble containers. A 
 themeable container is an element that can have custom classes attached which
```
Comment (hellogreg) [inline docs/theming/developers.md:51]: `attached which` → `attached, which`

File: docs/theming/developers.md [hunk 6]
```diff
@@ -156,15 +163,15 @@ rectangle.
 </svg>
 ```
 
-This approach _does not work_ with svg loaded through the `<img>` tag, or with 
-raster graphics, however another approach is in development which could help:
+This approach _does not work_ with SVG loaded through the `<img>` tag, or with
```
Comment (hellogreg) [inline docs/theming/developers.md:166]: `SVG` → `SVGs`

File: docs/theming/developers.md [hunk 7]
```diff
@@ -200,46 +207,47 @@ themable graphics
...
 ## How theming works
 
-We developed our theming system with web standards in mind, aiming for it to use
-CSS over JavaScript as much as possible. At the current state of the art, we
-found that some JavaScript is required for the system to work correctly, but in 
-the near  future, we expect to be able to remove all or nearly all of that 
-JavaScript, and have the theming system work entirely (or almost entirely) 
+We developed our theming system with web standards in mind, aiming to use
+CSS over JavaScript as much as possible. At the current state of browser technology,
+we found that some JavaScript is required for the system to work correctly.
```
Comment (hellogreg) [inline docs/theming/developers.md:212]: `we` → `we've`

File: docs/theming/developers.md [hunk 8]
```diff
@@ -200,46 +207,47 @@ themable graphics
...
+But shortly, we expect to be able to remove all or nearly all of that
```
Comment (hellogreg) [inline docs/theming/developers.md:213]: `shortly, we` → `we soon`

`all or nearly all of` → `all (or nearly all) of`
_This matches the "entirely (or almost entirely)" parenthetical in the next line._

File: docs/theming/developers.md [hunk 9]
```diff
@@ -200,46 +207,47 @@ themable graphics
...
+But shortly, we expect to be able to remove all or nearly all of that 
+JavaScript and have the theming system work entirely (or almost entirely) 
 through plain CSS. The following explains how the system currently operates, how
 element authors can hook into it, and what the future holds for theming.
 
 ### Current implementation - Context Protocol
 
-<abbr title="Red Hat Design System">RHDS</abbr>' Theming system is primarily 
-about styles. Tt currently relies on JavaScript to work, by way of the [context 
+<abbr title="Red Hat Design System">RHDS</abbr>’ theming system is primarily
```
Comment (hellogreg) [inline docs/theming/developers.md:220]: `RHDS’` → `RHDS’s `

File: docs/theming/developers.md [hunk 10]
```diff
@@ -200,46 +207,47 @@ themable graphics
...
-about styles. Tt currently relies on JavaScript to work, by way of the [context 
+<abbr title="Red Hat Design System">RHDS</abbr>’ theming system is primarily 
+about styles. It currently relies on JavaScript to work, via the [context 
 protocol][contextprotocol] developed by the web components community to support 
 passing data between components.
 
 Our system utilizes this protocol with the setting of the `color-palette` 
 attribute on a provider element which makes its context data (in our case, 
-`light` or `dark`) to it's children.  By doing so we can ensure accessible 
+`light` or `dark`) to its children. By doing so we ensure accessible
```
Comment (hellogreg) [inline docs/theming/developers.md:227]: I feel like a word is missing before "to its children." Is "available" the appropriate word here?

File: docs/theming/developers.md [hunk 11]
```diff
@@ -200,46 +207,47 @@ themable graphics
...
 
-<abbr title="Red Hat Design System">RHDS</abbr>' Theming system is primarily 
-about styles. Tt currently relies on JavaScript to work, by way of the [context 
+<abbr title="Red Hat Design System">RHDS</abbr>’ theming system is primarily 
+about styles. It currently relies on JavaScript to work, via the [context 
 protocol][contextprotocol] developed by the web components community to support 
 passing data between components.
 
 Our system utilizes this protocol with the setting of the `color-palette` 
 attribute on a provider element which makes its context data (in our case,
```
Comment (hellogreg) [inline docs/theming/developers.md:226]: `element which` → `element, which`

File: docs/theming/developers.md [hunk 12]
```diff
@@ -200,46 +207,47 @@ themable graphics
...
 <abbr title="Document Object Model">DOM</abbr> tree. 
 
-This is important not only for helping us maintain our own design guidelines, 
-but also for accessibility compliance and enabling great experiences for all our 
+This is important not only for helping us maintain our design guidelines but
+also for accessibility compliance and enabling great experiences for all our 
 users.
 
 The context protocol is enabled by two [reactive controllers][controllers]: the
 [provider controller][providersrc], and the [consumer controller][consumersrc].
```
Comment (hellogreg) [inline docs/theming/developers.md:236]: Delete the comma, since there are only two items in the list.

File: docs/theming/developers.md [hunk 13]
```diff
@@ -200,46 +207,47 @@ themable graphics
...
+on the needs of that particular element.
 
 #### Providers
 
 Custom Elements that implement the provider controller are elements that provide
-their own context, overriding that of their parent.  Elements such as
+their own context, overriding that of their parent. Elements such as
 `<rh-surface>`, `<rh-card>`, and `<rh-accordion>` are examples of such context
 providers. If a provider contains a set `color-palette` attribute, it will
 override any parent context, and pass its context on to its children.
```
Comment (hellogreg) [inline docs/theming/developers.md:248]: Delete the comma after "context."

File: docs/theming/developers.md [hunk 14]
```diff
@@ -272,14 +280,15 @@ Context consumers are elements that passively consume the parent context.
 Elements such as `<rh-cta>`, `<rh-badge>` and `<rh-tag>` are examples of
 consumers.
 
-To make your element a color context consumer,
+To make your element a color context consumer:
 
-1. First `import` the classMap Lit directive, and the consumer controller
+1. First `import` the [classMap Lit directive](https://lit.dev/docs/templates/directives/#classmap),
+   and the consumer controller.
 2. Then add the `@colorContextConsumer()` decorator to a private property of
```
Comment (hellogreg) [inline docs/theming/developers.md:287]: `on which` → `on, which`

File: docs/theming/developers.md [hunk 15]
```diff
@@ -272,14 +280,15 @@ Context consumers are elements that passively consume the parent context.
...
    `on` which is the type of `ColorTheme`.
-3. Add a classMap that implements the shadow class in your render method
-4. Use the theming tokens in your element's shadow styles, being sure.
-   to select the element which has the `.on.light`/`.on.dark` classes
+3. Add a classMap that implements the shadow class in your render method.
+4. Use the theming tokens in your element’s shadow styles, being sure
+   to select the element which has the `.on.light`/`.on.dark` classes.
```
Comment (hellogreg) [inline docs/theming/developers.md:291]: `which` → `that`

File: docs/theming/developers.md [hunk 16]
```diff
@@ -341,24 +350,24 @@ and applies values to the theming tokens, depending on the context received.
...
 
 In anticipation of this upcoming browser feature, we attempt to ensure that our
-theming system as implemented today using context can easily replaced with style
-queries in the near future.
+theming system as implemented today using context can easily be replaced with
+style queries in the near future.
 
-<rh-alert>Note, the examples in this section are hypothetical, the final markup
-and styles may not be the same</rh-alert>
+<rh-alert>Note: the examples in this section are hypothetical, the final markup
```
Comment (hellogreg) [inline docs/theming/developers.md:365]: `hypothetical, the` → `hypothetical. The`

File: docs/theming/developers.md [hunk 17]
```diff
@@ -377,7 +386,7 @@ And your document CSS sets the desired color context:
 ```
 
 And you would declare the import a shared stylesheet which activates the theming system,
```
Comment (hellogreg) [inline docs/theming/developers.md:388]: `which` → `that`

Comment (adamjohnson) [reply docs/theming/developers.md]: [8ae3170](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/8ae31707ea11f29ae967709bb95afebae9820f44)

File: docs/theming/developers.md (hunk 16 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: [8ae3170](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/8ae31707ea11f29ae967709bb95afebae9820f44)

File: docs/theming/developers.md (hunk 15 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: [8ae3170](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/8ae31707ea11f29ae967709bb95afebae9820f44)

File: docs/theming/developers.md (hunk 14 above)
Comment (adamjohnson) [reply docs/theming/developers.md:287]: Not seeing `on which` in this diff. Where are you suggesting this change?

File: docs/theming/developers.md (hunk 10 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: [a772217](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/a772217d3dc68bee1fa79a0781d8b31ed8b69886)

File: docs/theming/developers.md (hunk 9 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: [72faa86](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/72faa86569389ea1ae643d7ae84892a2094aeeec)

File: docs/theming/developers.md (hunk 8 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: [fe891c8](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/fe891c8237167f995351267d3e97915950f26d3a)

File: docs/theming/developers.md (hunk 7 above)
Comment (adamjohnson) [reply docs/theming/developers.md:212]: [917e417](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/917e4175b7f4667de89166632a801d9179979bd1)

File: docs/theming/developers.md (hunk 6 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: [372c601](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/372c601dfee5ccd30a81f465f74e8603bc3c473b)

File: docs/theming/developers.md [hunk 18]
```diff
@@ -139,8 +146,8 @@ two ways to approach this:
 
 ### Dynamic graphics
 
-Page authors can create dynamic graphics which respond to their surrounding 
-theme by using _inline SVG_ that references theme tokens. For example, this svg
+Page authors can create dynamic graphics that respond to their surrounding 
+theme by using _inline SVG_ that references theme tokens. For example, this SVG
```
[the remark this answers is not in the document]
Comment (adamjohnson) [reply docs/theming/developers.md]: [372c601](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/372c601dfee5ccd30a81f465f74e8603bc3c473b)

File: docs/theming/developers.md (hunk 5 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: [ead6a93](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/ead6a9333dfa84b5e52d37e67b74728d4edad97a)

File: docs/theming/developers.md (hunk 4 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: [9946521](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/9946521fde21d1067abffb3aaab574d91d35b73d)

File: docs/theming/developers.md (hunk 3 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: [8aa6731](https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/8aa6731ac3ce8c1716894fc135062a58ed031088)

File: docs/theming/developers.md (hunk 2 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/f29dd9c1ef1cc639b97eefc0eecc8d30f68b0d43

File: docs/theming/developers.md (hunk 11 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/8ae31707ea11f29ae967709bb95afebae9820f44

File: docs/theming/developers.md (hunk 13 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/8ae31707ea11f29ae967709bb95afebae9820f44

File: docs/theming/developers.md (hunk 12 above)
Comment (adamjohnson) [reply docs/theming/developers.md]: https://github.com/RedHat-UX/red-hat-design-system/pull/1937/commits/8ae31707ea11f29ae967709bb95afebae9820f44

**Review by zeroedin — APPROVED** [review]

**Pull request merged**